From 0d1b1689d578fcd9a32f27644f5a27acdd88bcf1 Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Fri, 17 May 2024 13:36:45 +0800
Subject: [PATCH 2/3] hdmitx only: support hdmitx only by dt overlay [1/3]

PD#SWPL-169477

Problem:
The yocto project does not discriminate dtb file
names when using HDMI only.

Solution:
Add a config, which the yocto project will open using HDMI only.

Verify:
local

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: If35a451324bb651277a37d693a9e2036fe216152
---
 board/amlogic/t7_an400_lpddr4x/t7_an400_lpddr4x.c | 4 ++--
 board/amlogic/t7_an400_lpddr4x_debian/Kconfig     | 5 +++++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/board/amlogic/t7_an400_lpddr4x/t7_an400_lpddr4x.c b/board/amlogic/t7_an400_lpddr4x/t7_an400_lpddr4x.c
index b6c9b5d63af..d2833732c91 100644
--- a/board/amlogic/t7_an400_lpddr4x/t7_an400_lpddr4x.c
+++ b/board/amlogic/t7_an400_lpddr4x/t7_an400_lpddr4x.c
@@ -521,13 +521,13 @@ int checkhw(char * name)
 	switch (ddr_size) {
 	case CONFIG_T7_4G_SIZE:
 		if (cpu_id.chip_rev == 0xA || cpu_id.chip_rev == 0xb) {
-			#ifdef CONFIG_HDMITX_ONLY
+			#if defined(CONFIG_HDMITX_ONLY) && !defined(CONFIG_AML_DT_OVERLAY)
 			strcpy(loc_name, "t7_a311d2_an400-hdmitx-only\0");
 			#else
 			strcpy(loc_name, "t7_a311d2_an400\0");
 			#endif
 		} else if (cpu_id.chip_rev == 0xC) {
-			#ifdef CONFIG_HDMITX_ONLY
+			#if defined(CONFIG_HDMITX_ONLY) && !defined(CONFIG_AML_DT_OVERLAY)
 			strcpy(loc_name, "t7c_a311d2_an400-hdmitx-only-4g\0");
 			#else
 			strcpy(loc_name, "t7c_a311d2_an400-4g\0");
diff --git a/board/amlogic/t7_an400_lpddr4x_debian/Kconfig b/board/amlogic/t7_an400_lpddr4x_debian/Kconfig
index dcbfc5144ad..376b5b8a701 100644
--- a/board/amlogic/t7_an400_lpddr4x_debian/Kconfig
+++ b/board/amlogic/t7_an400_lpddr4x_debian/Kconfig
@@ -27,4 +27,9 @@ config DTB_BIND_KERNEL
 config AML_REBOOT
        bool "reboot cmd"
        default y
+
+config AML_DT_OVERLAY
+    bool "use dts overlay"
+    default n if HDMITX_ONLY="true"
+
 endif
-- 
2.25.1

